client/src/components/Markdown/parse.test.js

```diff
@@ -0,0 +1,112 @@
+        });
+
+        it("should leave galaxy directives of same type with other labels alone", () => {
+            const input = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output=moo)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(result);
+        });
+
+        it("should leave galaxy directives of other types with same labels alone", () => {
+            const input = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(input=from)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(result);
+        });
+
+        it("should swap simple directives of specified type", () => {
+            const input = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output=from)\n```\n";
+            const output = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output=to)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(output);
+        });
+
+        it("should swap single quoted directives of specified type", () => {
+            const input = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output='from')\n```\n";
+            const output = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output=to)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(output);
+        });
+
+        it("should swap single quoted directives of specified type with extra args", () => {
+            const input =
+                "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(footer='cow', output='from', title=dog)\n```\n";
+            const output =
+                "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(footer='cow', output=to, title=dog)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(output);
+        });
+
+        it("should swap double quoted directives of specified type", () => {
+            const input = 'some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output="from")\n```\n';
+            const output = "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(output=to)\n```\n";
+            const result = replaceLabel(input, "output", "from", "to");
+            expect(result).toBe(output);
+        });
+
+        it("should swap double quoted directives of specified type with extra args", () => {
+            const input =
+                "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(footer='cow', output=\"from\", title=dog)\n```\n";
+            const output =
+                "some random\n`markdown content`\n```galaxy\nhistory_dataset_embedded(footer='cow', output=to, title=dog)\n```\n";
```

Comment:
I see in this test case (and others) that you're expecting the double quotes to be dropped. So, I'm assuming that the quotes dissappearance is intentional. It is a little jarring, though.

Reply:
I just tried this, and broke it, so I think we need to keep the double quotes as delimiters for defining the label.
initial label: input > added it to MD report
changed label to: inputtty > note change in MD report, without double quotes
changed label to inputty bimputty > note: with a space, the MD report, only has `inputty ` with a space, but without the second word `bimputty`

Reply:
It isn't expected to work that way - it was just a byproduct of the way it does work. I'll see if I can fix it for spaces in a rebase of your branch.